src/centralserver/registration-service/src/test/java/org/niis/xroad/centralserver/registrationservice/controller/RegistrationRequestControllerTest.java

```diff
@@ -0,0 +1,142 @@
+package org.niis.xroad.centralserver.registrationservice.controller;
+
+import ee.ria.xroad.common.OcspTestUtils;
+import ee.ria.xroad.common.SystemProperties;
+import ee.ria.xroad.common.TestCertUtil;
+import ee.ria.xroad.common.certificateprofile.SignCertificateProfileInfo;
+import ee.ria.xroad.common.certificateprofile.impl.EjbcaSignCertificateProfileInfo;
+import ee.ria.xroad.common.conf.globalconf.EmptyGlobalConf;
+import ee.ria.xroad.common.conf.globalconf.GlobalConf;
+import ee.ria.xroad.common.identifier.SecurityServerId;
+
+import org.bouncycastle.cert.ocsp.CertificateStatus;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.niis.xroad.centralserver.registrationservice.service.AdminApiService;
+import org.niis.xroad.centralserver.registrationservice.testutil.TestAuthCertRegRequest;
+import org.niis.xroad.centralserver.registrationservice.testutil.TestAuthRegRequestBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.ByteArrayInputStream;
+import java.security.KeyPairGenerator;
+import java.security.cert.X509Certificate;
+
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+public class RegistrationRequestControllerTest {
+    public static final String CONTENT_TYPE = "multipart/related; boundary=partboundary";
+
+    @Autowired
+    RegistrationRequestController controller;
+
+    @TestConfiguration
+    static class TestConfig {
+        @Bean
+        AdminApiService adminApiServiceImpl() {
+            return (serverId, address, certificate) -> 0;
+        }
+    }
+
+    @BeforeClass
+    public static void setup() {
+        System.setProperty(SystemProperties.CONFIGURATION_PATH, "build/resources/test/testconf");
+        GlobalConf.reload(new EmptyGlobalConf() {
+
+            @Override
+            public String getInstanceIdentifier() {
+                return "REST";
+            }
+
+            @Override
+            public int getOcspFreshnessSeconds(boolean smallestValue) {
+                return Integer.MAX_VALUE / 2;
+            }
+
+            @Override
+            public X509Certificate getCaCert(String instanceIdentifier, X509Certificate orgCert) {
+                return TestCertUtil.getCaCert();
+            }
+
+            @Override
+            public SignCertificateProfileInfo getSignCertificateProfileInfo(
+                    SignCertificateProfileInfo.Parameters parameters, X509Certificate cert) {
+                return new EjbcaSignCertificateProfileInfo(parameters);
+            }
+        });
+    }
+
+    @Test
+    public void shouldFailIfAuthSignatureIsInvalid() throws Exception {
```

Comment:
Maybe could consider adding more tests to verify the various logic of `ManagementRequestHandler`. Not sure what test cases would be the best bang for buck.

There is one specific test that was in the old implementation, and is not recreated as far as I can tell: "Should refuse to accept management request of different instance"